integrationtest/src/androidTest/java/com/swiftkey/cornedbeef/CoachMarkTestCase.java

```diff
@@ -107,7 +108,54 @@ public void run() {
         
         assertFalse(mCoachMark.isShowing());
     }
-    
+
+    /**
+     * Test that the popup disappears after the anchor view detaches if specified in the builder
+     */
+    public void testDismissAfterAnchorDetach() throws InterruptedException {
+        checkSetDismissOnAnchorDetach(true);
+        checkSetDismissOnAnchorDetach(false);
+    }
+
+    /**
+     * Helper method used in testDismissAfterAnchorDetach
+     *
+     * @param shouldAutoDismissOnDetach whether we should auto dismiss the coach mark
+     * @throws InterruptedException
+     */
+    private void checkSetDismissOnAnchorDetach(boolean shouldAutoDismissOnDetach) throws InterruptedException {
+        final View anchorView = new View(mActivity);
+        getInstrumentation().runOnMainSync(new Runnable() {
+            @Override
+            public void run() {
+                mActivity.setContentView(anchorView);
+            }
+        });
+
+        mCoachMark = new TestCoachMark.TestCoachMarkBuilder(
+                mActivity, anchorView, "spam spam spam")
+                .setDismissOnAnchorDetach(shouldAutoDismissOnDetach).build();
+
+        showCoachMark(getInstrumentation(), mCoachMark);
+
+        assertTrue(mCoachMark.isShowing());
+
+        Thread.sleep(2000);
```

Comment:
Do we need the long sleep? getInstrumentation().waitForIdleSync() may be enough.

Reply:
true, we don't need it